packages/isolate_bloc/lib/src/common/isolate/isolated_bloc_manager.dart

```diff
@@ -29,102 +33,118 @@ class IsolatedBlocManager {
-  static IsolatedBlocManager initialize(IsolatedConnector connector) {
-    return IsolatedBlocManager.instance = IsolatedBlocManager._(connector);
-  }
-
   /// Returns new bloc from cached in [_freeBlocs] or create new one.
-  IsolateBloc _getFreeBlocByType(Type type) {
-    IsolateBloc bloc;
+  IsolateBlocBase? _getFreeBlocByType(Type type) {
+    IsolateBlocBase? bloc;
     if (_freeBlocs.containsKey(type)) {
-      bloc = _freeBlocs.remove(type);
+      bloc = _freeBlocs.remove(type)!;
+    } else {
+      final blocCreator = _blocCreators[type];
+      if (blocCreator == null) {
+        print("Can't create IsolateBloc with type $type.\n"
+            "Maybe you forgot to register it?");
+      } else {
+        bloc = blocCreator.call();
+      }
+    }
+    if (bloc != null) {
+      _createdBlocs[bloc.id] = bloc;
+      return bloc;
     } else {
-      bloc = _blocCreators[type]();
+      return null;
     }
-    _createdBlocs[bloc.id] = bloc;
-    return bloc;
   }
 
-  Future<T> _getBloc<T extends IsolateBloc>() async {
+  Future<T> _getBloc<T extends IsolateBlocBase>() async {
     await _initializeCompleter.future;
-    // find every blocs with Type T
     final blocsT = _createdBlocs.values.whereType<T>();
     if (blocsT.isNotEmpty) {
       return blocsT.first;
     } else if (_freeBlocs.containsKey(T)) {
       return _freeBlocs[T] as T;
     } else {
-      return (_freeBlocs[T] = _blocCreators[T]()) as T;
+      final blocCreator = _blocCreators[T];
+      if (blocCreator == null) {
+        throw Exception("Failed to find BlocCreator for $T. Maybe you forget to `register` it?");
+      } else {
+        return _freeBlocs[T] = blocCreator() as T;
+      }
     }
   }
 
-  /// Use this function to get [IsolateBloc] in [Isolate].
-  /// To get bloc in UI [Isolate] use IsolateBlocProvider which returns [IsolateBlocWrapper].
+  /// Use this function to get [IsolateCubit] in [Isolate].
+  ///
+  /// To get bloc in UI [Isolate] use [IsolateBlocProvider] which returns [IsolateBlocWrapper].
   /// This function works this way: firstly it is wait for user's [Initializer] function
   /// secondly it is looks for created bloc with type BlocA. If it is finds any, so it
   /// returns this bloc's [IsolateBlocWrapper]. Else it is creates a new bloc and
   /// add to the pull of free blocs. So when UI will call `create()`, it will not create a new bloc but
   /// return free bloc from pull.
-  IsolateBlocWrapper<State>
-      getBlocWrapper<Bloc extends IsolateBloc<Object, State>, State>() {
-    IsolateBlocWrapper<State> wrapper;
-    Bloc isolateBloc;
+  IsolateBlocWrapper<State> getBlocWrapper<Bloc extends IsolateBlocBase<Object?, State>, State>() {
+    late IsolateBlocWrapper<State> wrapper;
+    Bloc? isolateBloc;
     _getBloc<Bloc>().then((bloc) {
       isolateBloc = bloc;
-      bloc.listen(wrapper.stateReceiver);
+
+      _createdBlocsSubscriptions[bloc.id] ??= [];
+      // ignore: invalid_use_of_protected_member
```

Comment:
What is ignored? Can't this comment be removed?

Reply:
This method annotated as `protected` so it is not private but shouldn't be used outside of it's sub classes. Some people just comment this kind of methods as `not part of public api`. We must do it because dart does not have a library or package visibility modifier